### ISSUE TITLE
Private assets in projects (not seen by the team)

### ISSUE THREAD

**Issue by cielecki** [10clouds/aiconsole#556]

**Comment by Globarti** [comment]
### Multi-Project team sharing experience with access control and asset control

**Summary: The idea here is to expand on the topic and build proper UI concept of AIConsole that caters:** 
- 1. Multi-project - collaborate inside multiple projects, workspaces that can be visible to users you want
- 2. Asset control (models, connections, materials, agents) - assets can be added to specific projects by users with asset modification rights
- 3. Access control - users can have different rights per project/workspace letting them Read, Write, or ad/modify the assets



### 1.Multi-project - inside your AIConsole main experience. 

User can access multiple workspaces that have multiple projects inside. I think AIC-user probably would be a member of a company. This company would have multiple teams like: Marketing team, Engineering team, and Management team. Each team have its workspace and in each workspace there are AIConsole projects that have assets (explained below) set-up in such way, that makes specific tasks for specific departments or jobs. 

As in NOTION there could be workspaces within your AIConsole with different set-up and data source according to the tasks you are doin in organisation. Example:
- Managers have a Project that is connected with **Salesforce reporting**
- Engineers have a Projects connected with **Confluence tasks**

Example:
<img width="510" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/7adccc64-7468-4035-b133-4f8d6af511c2">

Next thing is Public or Private projects/workspaces. In notion there are 3 tiers of sharing. 
- WORKSPACE - containing teams and its subpages 
- SHARED - are your projects that you and only chosen other users share
- PRIVATE - they are projects that ONLY YOU can see and use

Example:
<img width="268" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/35dd709a-adfd-4a5e-b8d3-da90e5041dab">

ADDITIONALLY -> Each project could be **DUPLICATED** to your personal space or shared spaces.



### 2. Asset control (Related to issue: https://github.com/10clouds/aiconsole/issues/560)
 models, connections, materials, agents could be set up per project/workspace. This would allow building JOB or DEPARTMENT specific projects that allows completing specific tasks at scale. 
 
 One of example could be that its toggled under each of the projects(visible only for users who have rights to edit)
<img width="307" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/b2ee1c45-f863-4bbf-be94-1c7678621de3">

As in Example from point 1. there could be different Projects/Workspacess with different assets:
- Managers have a Project that is connected with **Salesforce** and **Gmail** -> that allows to bring summaries of company 
- Engineers have a Projects connected with **Confluence tasks** and **Github Issues** that allows each user to both find and make new issues from the interface

EXAMPLE:
<img width="288" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/bf0dc61e-bf14-4274-97aa-674b033fde2d">


It could also be inproject or workspace settings 
<img width="652" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/b85f775f-2fb1-4bb8-b7c9-a586369f47ab">

**IMPORTANT** in this approeach each tab should be well thought threw to show in best way the added assets to the model.



### 3. Access control - each project / workspace should have its user permission rights tier:

Exemplary Permissions level / access level:
![image](https://github.com/10clouds/aiconsole/assets/1726746/0a753f6d-55d0-4463-94f6-ac14a6da72ca)

Each workspace / project should have access controll pane, that allows you to edit user rights levels and see what users are currnetly inside the project / workspace. Example from Notion:
<img width="663" alt="image" src="https://github.com/10clouds/aiconsole/assets/1726746/c3ce2a9b-cf8b-4856-9601-336f04eecca3">


### Ending note:
To sum up, I would love to have a complex Asset, Users and Worksapce/Projects management tools that bring a multi user experience to play. 

The most importantly this would allow application of the AIConsole to business, that values security and customisation.

**Comment by cielecki** [comment]
Solved by #716
